packages/hoppscotch-common/src/components/collections/index.vue

```diff
@@ -368,6 +373,29 @@ const currentUser = useReadonlyStream(
   platform.auth.getCurrentUserStream(),
   platform.auth.getCurrentUser()
 )
+
+/**
+ * Validates user authentication and token validity by making an API call
+ * @returns {Promise<object>} Authentication status with user existence and token validity
+ */
+const isValidUser = async () => {
+  if (currentUser.value) {
+    try {
+      const hasValidTokens = await verifyAuthTokens()
+      return {
+        valid: hasValidTokens,
+        error: SESSION_EXPIRED,
+      }
+    } catch (error) {
+      console.debug("Token validation failed:", error)
+      return { valid: false, error: SESSION_EXPIRED }
+    }
+  }
+
+  // allow user to perform actions without being logged in
+  return { valid: true, error: "" }
+}
```

Comment:
```suggestion
type ValidUserResponse = {
  valid: boolean
  error: string
}

/**
 * Validates user authentication and token validity by making an API call
 * @returns {Promise<ValidUserResponse>} Authentication status with user existence and token validity
 */
const isValidUser = async (): Promise<ValidUserResponse> => {
  // If user is logged in, verify tokens
  if (currentUser.value) {
    try {
      // Check if the user has valid tokens
      const hasValidTokens = await verifyAuthTokens()
      return {
        valid: hasValidTokens,
        error: SESSION_EXPIRED,
      }
    } catch (error) {
      console.debug("Token validation failed:", error)
      return { valid: false, error: SESSION_EXPIRED }
    }
  }

  // allow user to perform actions without being logged in
  return { valid: true, error: "" }
}
```

also we can move this to separate file and export this function from there